### PR TITLE
Breaking/nex 91/integrate npm item runner

### PR DESCRIPTION
**Pull request by oatymart**
Jira ticket: https://oat-sa.atlassian.net/browse/NEX-91
Vendorisation of item runner (reintegration)

**What I did:**
- deleted all js files & test files which were moved to `tao-item-runner-fe`
- registered new npm paths in php via install script & updater
- created `package.json` to install the new npm package
- enabled composer-npm-bridge to automatically install packages

**Prerequisites:**
- [x] https://github.com/oat-sa/tao-item-runner-fe/pull/6
- [x] release & publish on npmjs `tao-item-runner` @ 0.3.0
- [x] update above reference in this PR's package.json, & commit package-lock

**How to test:**
- Install this version of the `taoItems` extension in your TAO platform.
- Verify that the `@oat-sa/tao-item-runner` npm package was installed to `views/node_modules`.
- Verify that TAO backoffice and test delivery still work with no 404 errors (DEBUG mode).
- Verify that `npx grunt taoitemsbundle` bundle generation task completes, and site works (PROD mode)
- Verify that `npx grunt connect:dev taoitemstest` finds and runs the remaining tests

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+/views/node_modules
```
Comment (btamas) [inline .gitignore]: I think you should ignore all `node_modules`.

File: views/package.json [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  "version": "0.1.0",
+  "license": "GPL-2.0",
+  "description": "taoItems frontend modules",
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/oat-sa/extension-tao-items.git"
+  },
+  "dependencies": {
+    "@oat-sa/tao-item-runner": "0.3.0",
+    "@babel/polyfill": "^7.4.4"
```
Comment (btamas) [inline views/package.json]: Why do you need the `@babel/polyfill here`?

Comment (krampstudio) [reply views/package.json]: no it's not needrf

File: manifest.php [hunk 3]
```diff
@@ -54,7 +55,8 @@
 		),
 		'php'	=> array(
             CreateItemDirectory::class,
-            RegisterCategoryService::class
+            RegisterCategoryService::class,
+            RegisterNpmPaths::class
```
Comment (julien-sebire) [inline manifest.php]: You can add a trailing comma on every line in php arrays, this makes the adding of line a one-line diff...

File: scripts/install/RegisterNpmPaths.php [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+     */
+    public function __invoke($params)
+    {
+        $assetService = $this->getServiceManager()->get(AssetService::SERVICE_ID);
+        $taoItemsNpmDist = $assetService->getJsBaseWww('taoItems') . 'node_modules/@oat-sa/tao-item-runner/dist/';
+        $clientLibRegistry = ClientLibRegistry::getRegistry();
+        $clientLibRegistry->register('taoItems/assets', $taoItemsNpmDist . 'assets');
+        $clientLibRegistry->register('taoItems/runner', $taoItemsNpmDist . 'runner');
+        $clientLibRegistry->register('taoItems/scoring', $taoItemsNpmDist . 'scoring');
+        return \common_report_Report::createSuccess('extra paths for taoItems set up.');
```
Comment (julien-sebire) [inline scripts/install/RegisterNpmPaths.php]: Code style: add a blank line before `return` statement.

File: scripts/install/RegisterNpmPaths.php [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+use oat\oatbox\extension\InstallAction;
+use oat\tao\model\ClientLibRegistry;
+use oat\tao\model\asset\AssetService;
+
+/**
+ * Register the paths of frontend modules which are installed from npm packages
+ *
+ * @author Martin Nicholson <martin@taotesting.com>
+ */
+class RegisterNpmPaths extends InstallAction
```
Comment (julien-sebire) [inline scripts/install/RegisterNpmPaths.php]: Code style: add a blank line before `class` statement.

File: scripts/install/RegisterNpmPaths.php [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2019 (original work) Open Assessment Technologies SA;
+ *
```
Comment (julien-sebire) [inline scripts/install/RegisterNpmPaths.php]: Remove extra blank lines in comment.

File: manifest.php (hunk 3 above)
Comment (ferenckiss89) [reply manifest.php]: Fixed

File: scripts/install/RegisterNpmPaths.php (hunk 5 above)
Comment (ferenckiss89) [reply scripts/install/RegisterNpmPaths.php]: fixed

File: scripts/install/RegisterNpmPaths.php (hunk 6 above)
Comment (ferenckiss89) [reply scripts/install/RegisterNpmPaths.php]: fixed

**Pull request merged**